cli/src/main/java/keywhiz/cli/commands/RenameAction.java

```diff
@@ -16,26 +16,24 @@ public RenameAction(RenameActionConfig config, KeywhizClient keywhiz) {
   }
 
   @Override public void run() {
-    validate();
+    switch (config.resourceType) {
+      case "secret":
+        renameSecret();
+        break;
+      default:
+        throw new IllegalArgumentException(
+            String.format(
+                "Unsupported resource type %s. Only the secret resource type can be renamed.",
+                config.resourceType));
+    }
```

Comment:
If we ever add other types this should be changed to list out some enum values, but I thought this was a bit more user-friendly for now.

Reply:
I think that's a good call